### ISSUE TITLE
Document make targets

### ISSUE THREAD

**Issue by nilsvu** [sxs-collaboration/spectre#903]
# Feature request:

The Travis documentation test repeats Doxygen warnings and errors at the end of the log, but the `make doc` target doesn't. This would be useful.

### Component:

- [x] Build system

**Comment by wthrowe** [comment]
Is `make doc-check` close enough to what you want?

**Comment by nilsvu** [comment]
Yep that's what I was looking for. We should document these useful make targets :)

**Comment by nilsdeppe** [comment]
There's a `make list` to show all the targets. Is that not sufficient?

**Comment by nilsvu** [comment]
I think we should have a list in the documentation with a description of useful targets and how to use them, e.g. for the various `doc`, `clang-tidy` and `iwyu` targets (they are somewhat documented in the Travis CI dev guide).

**Comment by nilsdeppe** [comment]
Sounds reasonable. Should we have that as a page that is linked to from the installation pages? Another question: could the list be autogenerated somehow using CMake?

Edit:
- With Makefile generators it's quite easy to generate a list of all the targets using `cmake --build /PATH/TO/BUILD/DIR --target help`. We could parse this list into something useful. For example, we could have a list of all the test targets, library targets, "special targets" (like IWYU, clang tidy, etc.), etc.
